tests/test_simplify.py

```diff
@@ -40,6 +41,12 @@ def test_simplify():
     assert quickTest("x/y + x/x + x/x^2 + x^2/x = x/y^2 + x^2/y + x - 1", simplifyEquation) == "xy^(-1)+2.0+x^(-1.0)-xy^(-2.0)-x^(2.0)y^(-1)=0"
 
 
+def test_preprocess_simplification():
+
+    assert preprocessSimplification("4+(4+(4+1))") == "4+4+4+1"
+    assert preprocessSimplification("4-(4+(4-1))") == "4-4-4+1"
```

Comment:
Input cases with * and / might not return the correct result.

Reply:
yeah I checked that and that's why I left a note stating that this pr does not simplify the input equation. It is just a temporary fix for the input case that was in the issue referenced here.

Reply:
I am talking about a case like `4-(4*(4-1))` which will reduce to `4-4*4+1`.

Reply:
Okay I am looking into this. Thank you.